backend/pharmacy/api/views.py

```diff
@@ -20,42 +28,74 @@ def get_token(cls, user):
+@method_decorator(cache_page(60), name='get')
+# A method decorator to vary on the cookie
+@method_decorator(vary_on_cookie, name='get')
+# A method decorator to vary on the headers
+# @method_decorator(vary_on_headers, name='get')
+# Allow only authenticated users to access this view
 @permission_classes([IsAuthenticated])
-def getData(request):
-    medical = Medical.objects.all()
-    serializer = MedicalSerializer(medical, many=True)
-    return Response(serializer.data)
+class Data(APIView):
+    def getObject(self, pk):
+        try:
+            return Medical.objects.filter(pk=pk)
+        except Medical.DoesNotExist:
+            return Http404
 
-@api_view(['POST'])
-@permission_classes([IsAuthenticated])
-def postData(request):
-    serializer = MedicalSerializer(data=request.data)
-    if serializer.is_valid():
-        serializer.save()
+            
+    def get(self, request, pk):
+        medical = self.getObject(pk)
+        serializer = MedicalSerializer(medical, many=True)
         return Response(serializer.data)
-    return Response(serializer.errors)
+    
+    def put(self, request, pk):
+        medical = self.getObject(pk)
+        # Ensure that the user is the owner of the medical
+        if medical[0].user.id != request.user.id:
+            return Response("HTTP 403 Forbidden", status=403)
+        serializer = MedicalSerializer(medical, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=400)
+    
+    def delete(self, request, pk):
+        medical = self.getObject(pk)
+        # Ensure that the user is the owner of the medical
+        if medical[0].user.id != request.user.id:
+            return Response("HTTP 403 Forbidden", status=403)
+        medical.delete()
+        return Response("Deleted", status=200)
+    
 
-@api_view(['PUT'])
 @permission_classes([IsAuthenticated])
-def updateData(request):
-    medicalId = request.data['medicalId']
-    medical = Medical.objects.get(medicalId)
-    serializer = MedicalSerializer(instance=medical, data=request.data)
-    if serializer.is_valid():
-        serializer.save()
+class DataList(APIView):
+    def get(self, request, format=None):
+        medical = Medical.objects.all()
+        serializer = MedicalSerializer(medical, many=True)
         return Response(serializer.data)
-    return Response(serializer.errors)
+
+    def post(self, request):
+        serializer = MedicalSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=201)
+        return Response(serializer.errors, status=400)
 
 
-@api_view(['GET'])
-def getRoutes(request):
-    routes = [
-        'api/token',
-        'api/token/refresh',
-    ]
-    return Response(routes)
\ No newline at end of file
+@permission_classes([IsAuthenticated])
+class User(APIView):
```

Comment:
I have made this class for testing purpose but it can be used later in our application to show user details....